File: my-folder/0134-gas-station/solution.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int totalFuel=0;
        int totalCost=0;
        int n = gas.length;
        for(int i=0;i<n;i++){
            totalFuel+=gas[i];
            totalCost+=cost[i];
        }
        if(totalFuel < totalCost)
            return -1;
        int start=0;
        int currFuel=0;
        for(int i=0;i<n;i++){
            if(currFuel<0){
                currFuel=0;
                start=i;
            }
            currFuel += gas[i]-cost[i];
        }
        return start;
    }
}

File: my-folder/0134-gas-station/solution.py
class Solution:
    # Greedy solution
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        if sum(gas) < sum(cost):
            return -1

        total_gas = 0
        res_idx = 0
        for i in range(len(gas)):
            total_gas += gas[i] - cost[i]

            if total_gas < 0:
                total_gas = 0
                res_idx = i + 1

        return res_idx

# Time -> O(n)
# Space -> O(1)       
        
